db/infinispan/src/main/java/org/commonjava/indy/infinispan/data/InfinispanStoreDataManager.java

```diff
@@ -50,6 +53,10 @@
     @StoreDataCache
     private CacheHandle<StoreKey, ArtifactStore> stores;
 
+    @Inject
+    @StoreByPkgCache
+    private CacheHandle<String, Map<StoreType, Set<StoreKey>>> storesByPkg;
```

Comment:
I think this can really only work if it's never persisted or clustered. That might cause problems if we try to deploy in a multi-node scenario.

Otherwise, I think the write latency on this cache combined with the synchronization of all store writes will crush performance. I could be misreading that, maybe I'm missing something.

Reply:
Currently this cache is a pure in-memory cache, whose content is built up from the store cache during each startup. I agree that this can not work well in multi-node scenario. If we want this to work in distributed world, persistence or distributed cache is always needed.